h2o-mapreduce-generic/src/main/java/water/hadoop/DelegationTokenRefresher.java

```diff
@@ -0,0 +1,152 @@
+package water.hadoop;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.security.Credentials;
+import org.apache.hadoop.security.UserGroupInformation;
+import water.H2O;
+import water.MRTask;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.security.PrivilegedExceptionAction;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static water.hadoop.h2omapper.*;
+
+public class DelegationTokenRefresher implements Runnable {
+  
+  public static final String KEYTAB_FILE = "hive_keytab";
+  private static final String KEYTAB_PATH = "./" + KEYTAB_FILE;
+
+  public static void setup(Configuration conf) {
+    String hiveHost = conf.get(H2O_HIVE_HOST);
+    String hivePrincipal = conf.get(H2O_HIVE_PRINCIPAL);
+    String authPrincipal = conf.get(H2O_AUTH_PRINCIPAL);
+
+    if (authPrincipal != null && hiveHost != null && hivePrincipal != null) {
+      new DelegationTokenRefresher(conf, authPrincipal, hiveHost, hivePrincipal).start();
+    }
+  }
+
+  private final ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor(
+      new ThreadFactoryBuilder().setDaemon(true).setNameFormat("delegation-token-refresher-%d").build()
+  );
+
+  private final Configuration conf;
+  private final String authPrincipal;
+  private final String hiveHost;
+  private final String hivePrincipal;
+
+  private final HiveTokenGenerator hiveTokenGenerator = new HiveTokenGenerator();
+
+  public DelegationTokenRefresher(
+      Configuration conf,
+      String authPrincipal,
+      String hiveHost,
+      String hivePrincipal
+  ) {
+    this.conf = conf;
+    this.authPrincipal = authPrincipal;
+    this.hiveHost = hiveHost;
+    this.hivePrincipal = hivePrincipal;
+  }
+
+  public void start() {
+    if (hiveTokenGenerator.isHiveDriverPresent()) {
+      executor.scheduleAtFixedRate(this, 0, 1, TimeUnit.MINUTES);
```

Comment:
Does it makes sense to make this configurable ?

Reply:
I am not sure if that would be useful to anyone.